### PR TITLE
Ensure that an EnvDTE.CodeParameter.Name returns the correct value for parameters defined in metadata

### LINKED ISSUE #7210: Code Model: CodeParameter2 in DTE code model has no value for Name after Update 1
This appears to have regressed in Update 1.

### PR DESCRIPTION
**Pull request by DustinCampbell**
Fixes #7210

The issue is that the symbol display system will not return a name for a parameter unless SymbolDisplayParameterOptions.IncludeName is set. This behavior seems like it might ultimately be a bug in symbol display as that option is intended to control the display the parameters owned by the targeted symbol, but probably shouldn't affect the display of the targeted symbol. However, rather than updating symbol display (which would potentially have larger downstream impact), we'll just include the option in this case.

#7211 has been filed for the symbol display issue.

Tagging @dotnet/roslyn-ide for review

### CODE REVIEW

**Comment by DustinCampbell** [comment]
test vsi please

**Comment by Pilchie** [comment]
Thanks for jumping on this!  :+1:

File: src/VisualStudio/Core/Test/CodeModel/CSharp/ExternalCodeFunctionTests.vb [hunk 1]
```diff
@@ -0,0 +1,51 @@
+﻿Imports System.Threading.Tasks
```
Comment (brettfo) [inline src/VisualStudio/Core/Test/CodeModel/CSharp/ExternalCodeFunctionTests.vb]: copyright

File: src/VisualStudio/Core/Test/CodeModel/VisualBasic/ExternalCodeFunctionTests.vb [hunk 2]
```diff
@@ -0,0 +1,47 @@
+﻿Imports System.Threading.Tasks
```
Comment (brettfo) [inline src/VisualStudio/Core/Test/CodeModel/VisualBasic/ExternalCodeFunctionTests.vb]: copyright

File: src/VisualStudio/Core/Test/CodeModel/CSharp/ExternalCodeParameterTests.vb [hunk 3]
```diff
@@ -0,0 +1,113 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
```
Comment (brettfo) [inline src/VisualStudio/Core/Test/CodeModel/CSharp/ExternalCodeParameterTests.vb]: No project file updates for added files?

Comment (brettfo) [inline src/VisualStudio/Core/Test/CodeModel/CSharp/ExternalCodeParameterTests.vb]: Nm, just found it :)

File: src/VisualStudio/Core/Test/CodeModel/VisualBasic/ExternalCodeFunctionTests.vb (hunk 2 above)
Comment (DustinCampbell) [inline src/VisualStudio/Core/Test/CodeModel/VisualBasic/ExternalCodeFunctionTests.vb]: Thanks!

File: src/VisualStudio/Core/Test/CodeModel/CSharp/ExternalCodeFunctionTests.vb (hunk 1 above)
Comment (DustinCampbell) [inline src/VisualStudio/Core/Test/CodeModel/CSharp/ExternalCodeFunctionTests.vb]: Thanks!

File: src/VisualStudio/Core/Test/CodeModel/CSharp/CodeParameterTests.vb [hunk 4]
```diff
@@ -102,6 +102,55 @@ class C : System.Console
 
 #End Region
 
+#Region "Name tests"
+
+        <ConditionalWpfFact(GetType(x86)), Trait(Traits.Feature, Traits.Features.CodeModel)>
```
Comment (dpoeschl) [inline src/VisualStudio/Core/Test/CodeModel/CSharp/CodeParameterTests.vb]: Does this need to be a wpf fact?

Comment (DustinCampbell) [inline src/VisualStudio/Core/Test/CodeModel/CSharp/CodeParameterTests.vb]: Please file an issue

**Comment by dpoeschl** [comment]
The description makes it seem like the only change was to add .IncludeName, but there's also new handling of CodeParameters in AsExternal. Might be worth including that part in the commit message for future archaeology.

File: src/VisualStudio/Core/Test/CodeModel/CSharp/CodeParameterTests.vb (hunk 4 above)
Comment (dpoeschl) [inline src/VisualStudio/Core/Test/CodeModel/CSharp/CodeParameterTests.vb]: https://github.com/dotnet/roslyn/issues/7214

**Comment by dpoeschl** [comment]
:+1:

**Comment by DustinCampbell** [comment]
retest prtest/win/dbg/unit64

**Comment by DustinCampbell** [comment]
test vsi please

**Comment by DustinCampbell** [comment]
retest prtest/win/dbg/unit64 please

**Comment by jasonmalinowski** [comment]
:+1:

**Pull request merged**